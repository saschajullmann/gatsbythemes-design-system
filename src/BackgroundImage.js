import styled from '@emotion/styled'
import PropTypes from 'prop-types'
import { width } from 'styled-system'
import theme from './theme'

const image = props =>
  props.image ? { backgroundImage: `url(${props.image})` } : null

const height = props => (props.height ? { height: props.height } : null)

const BackgroundImage = styled.div`
  background-position: center;
  background-size: cover;
  background-repeat: no-repeat;
  background-color: ${props => props.theme.colors.gray};
  ${image} ${height} ${width};
`
const numberStringOrArray = PropTypes.oneOfType([
  PropTypes.number,
  PropTypes.string,
  PropTypes.array
])

BackgroundImage.propTypes = {
  /** background-image url */
  image: PropTypes.string,
  width: numberStringOrArray
}

BackgroundImage.defaultProps = {
  theme: theme
}

BackgroundImage.displayName = 'BackgroundImage'

export default BackgroundImage
