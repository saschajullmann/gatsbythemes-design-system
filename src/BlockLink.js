import styled from '@emotion/styled'
import Link from './Link'

const BlockLink = styled(Link)`
  display: block;
  color: inherit;
  text-decoration: none;
`

BlockLink.displayName = 'BlockLink'

export default BlockLink
