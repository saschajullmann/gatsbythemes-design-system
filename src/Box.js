import styled from '@emotion/styled'
import { space, width, color, textAlign } from 'styled-system'
import PropTypes from 'prop-types'
import theme from './theme'

const Box = styled.div`
  ${space} ${width} ${color} ${textAlign}
`

Box.displayName = 'Box'

Box.defaultProps = {
  theme: theme
}

Box.propTypes = {
  ...space.propTypes,
  ...width.propTypes,
  ...color.propTypes,
  ...textAlign.propTypes
}

export default Box
