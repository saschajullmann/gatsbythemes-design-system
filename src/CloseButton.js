import React from 'react'
import IconButton from './IconButton'
import PropTypes from 'prop-types'
import theme from './theme'

const CloseButton = props => <IconButton {...props} name="close" />

CloseButton.defaultProps = {
  size: 24,
  title: 'close'
}

CloseButton.propTypes = {
  onClick: PropTypes.func,
  size: PropTypes.number,
  title: PropTypes.string
}

CloseButton.displayName = 'CloseButton'

export default CloseButton
