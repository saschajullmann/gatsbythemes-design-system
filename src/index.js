export { default as Absolute } from './Absolute'
export { default as BackgroundImage } from './BackgroundImage'
export { default as Badge } from './Badge'
export { default as Banner } from './Banner'
export { default as BlockLink } from './BlockLink'
export { default as Box } from './Box'
export { default as Button } from './Button'
export { default as Card } from './Card'
export { default as CloseButton } from './CloseButton'
export { default as Container } from './Container'
export { default as Divider } from './Divider'
export { default as Flag } from './Flag'
export { default as Flex } from './Flex'
export { default as GreenButton } from './GreenButton'
export { default as Heading } from './Heading'
export { default as Hide } from './Hide'
export { default as Hug } from './Hug'
export { default as Icon } from './Icon'
export { default as IconButton } from './IconButton'
export { default as IconField } from './IconField'
export { default as Image } from './Image'
export { default as Input } from './Input'
export { default as InputField } from './FormField'
export { default as FormField } from './FormField'
export { default as InputGroup } from './InputGroup'
export { default as Label } from './Label'
export { default as Link } from './Link'
export { default as OutlineButton } from './OutlineButton'
export { default as Radio } from './Radio'
export { default as RatingBadge } from './RatingBadge'
export { default as RedButton } from './RedButton'
export { default as Relative } from './Relative'
export { default as Stamp } from './Stamp'
export { default as Select } from './Select'
export { default as Text } from './Text'
export { default as Tooltip } from './Tooltip'
export { default as ToggleBadge } from './ToggleBadge'
export { default as Truncate } from './Truncate'
export { default as Checkbox } from './Checkbox'

export { default as theme } from './theme'
export { default as ThemeProvider } from './ThemeProvider'

export * from './theme'