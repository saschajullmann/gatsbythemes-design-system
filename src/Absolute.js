import styled from '@emotion/styled'
import Box from './Box'
import { top, right, bottom, left, zIndex } from 'styled-system'

const Absolute = styled(Box)`
  position: absolute;
  ${top} ${bottom} ${left} ${right}
  ${zIndex}
`

Absolute.propTypes = {
  ...top.propTypes,
  ...right.propTypes,
  ...bottom.propTypes,
  ...left.propTypes,
  ...zIndex.propTypes
}

Absolute.displayName = 'Absolute'

export default Absolute
