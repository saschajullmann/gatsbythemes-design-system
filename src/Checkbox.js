import React from 'react'
import styled from '@emotion/styled'
import { css } from '@emotion/core'
import PropTypes from 'prop-types'
import theme from './theme'
import Icon from './Icon'
import Box from './Box'

const Checkbox = props => {
  const { disabled, size } = props
  return (
    <CheckBoxWrapper disabled={disabled}>
      <StyledInput type="checkbox" {...props} />
      <Icon name="boxChecked" size={size} data-name="checked" />
      <Icon name="boxEmpty" size={size} data-name="empty" />
    </CheckBoxWrapper>
  )
}

const CheckBoxWrapper = styled(Box)`
  display: inline-block;
  position: relative;
  vertical-align: middle;
  cursor: pointer;
  color: ${props =>
    props.disabled ? props.theme.colors.borderGray : props.theme.colors.gray};

  svg[data-name='checked'] {
    display: none;
  }

  > input:checked {
    & ~ svg[data-name='checked'] {
      display: inline-block;
      color: ${props =>
    props.disabled
      ? props.theme.colors.borderGray
      : props.theme.colors.blue};
    }

    & ~ svg[data-name='empty'] {
      display: none;
    }
  }
`

CheckBoxWrapper.defaultProps = {
  theme
}

const StyledInput = styled.input`
  appearance: none;
  opacity: 0;
  position: absolute;
  z-index: 0;
`

Checkbox.displayName = 'Checkbox'

Checkbox.propTypes = {
  id: PropTypes.string.isRequired,
  size: PropTypes.number,
  onChange: PropTypes.func.isRequired
}

Checkbox.defaultProps = {
  size: 20
}

export default Checkbox
