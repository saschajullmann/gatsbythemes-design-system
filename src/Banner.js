import React from 'react'
import Box from './Box'
import Flex from './Flex'
import Text from './Text'
import Icon from './Icon'
import CloseButton from './CloseButton'
import Heading from './Heading'
import PropTypes from 'prop-types'

const bannerColors = {
  green: {
    backgroundColor: 'green',
    color: 'white',
    icon: 'success'
  },
  lightGreen: {
    backgroundColor: 'lightGreen',
    color: 'darkGreen',
    icon: 'success'
  },
  red: {
    backgroundColor: 'red',
    color: 'white',
    icon: 'warning'
  },
  lightRed: {
    backgroundColor: 'lightRed',
    color: 'darkRed',
    icon: 'warning'
  },
  orange: {
    backgroundColor: 'orange',
    color: 'white',
    icon: 'attention'
  },
  blue: {
    backgroundColor: 'blue',
    color: 'white',
    icon: 'information'
  },
  lightBlue: {
    backgroundColor: 'lightBlue',
    color: 'darkBlue',
    icon: 'information'
  }
}

const Banner = props => {
  const bannerColor = bannerColors[props.bg] || {}
  const icon = props.iconName || bannerColor.icon

  return (
    <Box
      {...props}
      bg={bannerColor.backgroundColor || props.bg}
      color={bannerColor.color || props.color}
    >
      <Flex justifyContent="space-between" alignItems="flex-start">
        {!!icon && !!props.showIcon && (
          <Icon name={icon} mr={2} size={24} mt="-2px" />
        )}
        <Box w={1}>
          <Text textAlign={props.textAlign}>
            <Heading.h5>{props.header}</Heading.h5>
            <Text.span fontSize={1}>{props.text}</Text.span>
            {props.children}
          </Text>
        </Box>
        {!!props.onClose && (
          <CloseButton
            onClick={props.onClose}
            ml={2}
            size={24}
            title="close"
            mt="-2px"
          />
        )}
      </Flex>
    </Box>
  )
}

Banner.displayName = 'Banner'

Banner.propTypes = {
  header: PropTypes.string,
  iconName: PropTypes.string,
  onClose: PropTypes.func,
  showIcon: PropTypes.bool,
  text: PropTypes.node,
  textAlign: PropTypes.string
}

Banner.defaultProps = {
  bg: 'green',
  textAlign: 'left',
  showIcon: true
}

export default Banner
