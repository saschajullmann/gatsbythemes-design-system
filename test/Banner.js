import React from 'react'
import renderer from 'react-test-renderer'
import { render, cleanup } from 'react-testing-library'
import { Banner, Text, theme } from '../src'

afterEach(cleanup)

describe('Banner', () => {
  test('renders with no props other than theme', () => {
    const json = renderer.create(<Banner />).toJSON()
    expect(json).toMatchSnapshot()
  })

  test('renders with custom iconName and size', () => {
    const json = renderer
      .create(<Banner iconName="star" iconSize={20} />)
      .toJSON()
    expect(json).toMatchSnapshot()
  })

  test('renders with text string', () => {
    const json = renderer
      .create(
        <Banner
          header="Header"
          text="Text"
          iconName="star"
          iconSize={20}
          theme={theme}
        />
      )
      .toJSON()
    expect(json).toMatchSnapshot()
  })

  test('renders with text node', () => {
    const json = renderer
      .create(
        <Banner
          header="Header"
          text={<Text>Text</Text>}
          iconName="star"
          iconSize={20}
          theme={theme}
        />
      )
      .toJSON()
    expect(json).toMatchSnapshot()
  })

  test('renders with blue bg', () => {
    const json = renderer.create(<Banner bg="blue" />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('background-color', theme.colors.blue)
    expect(json).toHaveStyleRule('color', theme.colors.white)
  })

  test('renders with green bg', () => {
    const json = renderer.create(<Banner bg="green" />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('background-color', theme.colors.green)
    expect(json).toHaveStyleRule('color', theme.colors.white)
  })

  test('renders with orange bg', () => {
    const json = renderer.create(<Banner bg="orange" />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('background-color', theme.colors.orange)
    expect(json).toHaveStyleRule('color', theme.colors.white)
  })

  test('renders with red bg', () => {
    const json = renderer.create(<Banner bg="red" />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('background-color', theme.colors.red)
    expect(json).toHaveStyleRule('color', theme.colors.white)
  })

  test('renders with lightBlue bg', () => {
    const json = renderer.create(<Banner bg="lightBlue" />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('background-color', theme.colors.lightBlue)
    expect(json).toHaveStyleRule('color', theme.colors.darkBlue)
  })

  test('renders with lightGreen bg', () => {
    const json = renderer.create(<Banner bg="lightGreen" />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('background-color', theme.colors.lightGreen)
    expect(json).toHaveStyleRule('color', theme.colors.darkGreen)
  })

  test('renders with lightRed bg', () => {
    const json = renderer.create(<Banner bg="lightRed" />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('background-color', theme.colors.lightRed)
    expect(json).toHaveStyleRule('color', theme.colors.darkRed)
  })

  test('renders close button if onClose func is provided', () => {
    const { container } = render(<Banner onClose={() => {}} />)
    const closeButton = container.querySelector('button')
    expect(closeButton).toBeTruthy()
  })

  test('does render blue left-hand icon by default', () => {
    const { container } = render(<Banner bg="blue" />)
    const icon = container.querySelector('svg')
    // todo: check svg path/attrs for correct icon
    expect(icon).toBeTruthy()
  })

  // these won't be needed if Banner is changed to a more composable API
  test.skip('does render green left-hand icon by default', () => {
    const { container } = render(<Banner bg="green" />)
    const icon = container.querySelector('svg')
    // name=success
    expect(icon).toBeTruthy()
  })

  test.skip('does render orange left-hand icon by default', () => {
    const { container } = render(<Banner bg="orange" />)
    const icon = container.querySelector('svg')
    // name=attention
    expect(icon).toBeTruthy()
  })

  test.skip('does render red left-hand icon by default', () => {
    const { container } = render(<Banner bg="red" />)
    const icon = container.querySelector('svg')
    // name=warning
    expect(icon).toBeTruthy()
  })

  test('does not render blue left-hand icon if showIcon is false', () => {
    const { container } = render(<Banner bg="blue" showIcon={false} />)
    const icon = container.querySelector('svg')
    expect(icon).toBeFalsy()
  })

  test('accepts non-preset colors', () => {
    const json = renderer.create(<Banner bg="gray" />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('background-color', theme.colors.gray)
  })

  test('renders content from children props', () => {
    const json = renderer
      .create(
        <Banner>
          <span>123</span>
        </Banner>
      )
      .toJSON()
    expect(json).toMatchSnapshot()
  })
})
