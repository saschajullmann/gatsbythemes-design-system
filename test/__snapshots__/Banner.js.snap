// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner accepts non-preset colors 1`] = `
.c0 {
  background-color: #4f6f8f;
  text-align: left;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c2 {
  text-align: left;
}

.c4 {
  font-size: 14px;
}

.c3 {
  font-size: 24px;
  font-weight: 700;
  line-height: 1.25;
  margin: 0px;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className=""
    >
      <div
        className="c2"
      >
        <h5
          className="c3"
        />
        <span
          className="c4"
          fontSize={1}
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Banner renders content from children props 1`] = `
.c0 {
  color: #fff;
  background-color: #0a0;
  text-align: left;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c3 {
  text-align: left;
}

.c5 {
  font-size: 14px;
}

.c2 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  margin-right: 8px;
  margin-top: -2px;
}

.c4 {
  font-size: 24px;
  font-weight: 700;
  line-height: 1.25;
  margin: 0px;
}

<div
  className="c0"
  color="white"
>
  <div
    className="c1"
  >
    <svg
      className="c2"
      fill="currentcolor"
      height={24}
      viewBox="0 0 24 24"
      width={24}
    >
      <path
        d="M12 2C6.5 2 2 6.5 2 12s4.5 10 10 10 10-4.5 10-10S17.5 2 12 2zm-1.7 14.5L6 12.2 7.2 11l3.1 3.1 6.5-6.5L18 8.8l-7.7 7.7z"
      />
    </svg>
    <div
      className=""
    >
      <div
        className="c3"
      >
        <h5
          className="c4"
        />
        <span
          className="c5"
          fontSize={1}
        />
        <span>
          123
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Banner renders with blue bg 1`] = `
.c0 {
  color: #fff;
  background-color: #007aff;
  text-align: left;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c3 {
  text-align: left;
}

.c5 {
  font-size: 14px;
}

.c2 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  margin-right: 8px;
  margin-top: -2px;
}

.c4 {
  font-size: 24px;
  font-weight: 700;
  line-height: 1.25;
  margin: 0px;
}

<div
  className="c0"
  color="white"
>
  <div
    className="c1"
  >
    <svg
      className="c2"
      fill="currentcolor"
      height={24}
      viewBox="0 0 24 24"
      width={24}
    >
      <path
        d="M12 2C6.5 2 2 6.5 2 12s4.5 10 10 10 10-4.5 10-10S17.5 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
      />
    </svg>
    <div
      className=""
    >
      <div
        className="c3"
      >
        <h5
          className="c4"
        />
        <span
          className="c5"
          fontSize={1}
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Banner renders with custom iconName and size 1`] = `
.c0 {
  color: #fff;
  background-color: #0a0;
  text-align: left;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c3 {
  text-align: left;
}

.c5 {
  font-size: 14px;
}

.c2 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  margin-right: 8px;
  margin-top: -2px;
}

.c4 {
  font-size: 24px;
  font-weight: 700;
  line-height: 1.25;
  margin: 0px;
}

<div
  className="c0"
  color="white"
>
  <div
    className="c1"
  >
    <svg
      className="c2"
      fill="currentcolor"
      height={24}
      viewBox="0 0 24 24"
      width={24}
    >
      <path
        d="M12 17.8l6.2 3.7-1.6-7L22 9.7l-7.2-.6L12 2.5 9.2 9.1 2 9.7l5.5 4.8-1.6 7 6.1-3.7z"
      />
    </svg>
    <div
      className=""
    >
      <div
        className="c3"
      >
        <h5
          className="c4"
        />
        <span
          className="c5"
          fontSize={1}
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Banner renders with green bg 1`] = `
.c0 {
  color: #fff;
  background-color: #0a0;
  text-align: left;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c3 {
  text-align: left;
}

.c5 {
  font-size: 14px;
}

.c2 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  margin-right: 8px;
  margin-top: -2px;
}

.c4 {
  font-size: 24px;
  font-weight: 700;
  line-height: 1.25;
  margin: 0px;
}

<div
  className="c0"
  color="white"
>
  <div
    className="c1"
  >
    <svg
      className="c2"
      fill="currentcolor"
      height={24}
      viewBox="0 0 24 24"
      width={24}
    >
      <path
        d="M12 2C6.5 2 2 6.5 2 12s4.5 10 10 10 10-4.5 10-10S17.5 2 12 2zm-1.7 14.5L6 12.2 7.2 11l3.1 3.1 6.5-6.5L18 8.8l-7.7 7.7z"
      />
    </svg>
    <div
      className=""
    >
      <div
        className="c3"
      >
        <h5
          className="c4"
        />
        <span
          className="c5"
          fontSize={1}
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Banner renders with lightBlue bg 1`] = `
.c0 {
  color: #049;
  background-color: #e8f2ff;
  text-align: left;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c3 {
  text-align: left;
}

.c5 {
  font-size: 14px;
}

.c2 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  margin-right: 8px;
  margin-top: -2px;
}

.c4 {
  font-size: 24px;
  font-weight: 700;
  line-height: 1.25;
  margin: 0px;
}

<div
  className="c0"
  color="darkBlue"
>
  <div
    className="c1"
  >
    <svg
      className="c2"
      fill="currentcolor"
      height={24}
      viewBox="0 0 24 24"
      width={24}
    >
      <path
        d="M12 2C6.5 2 2 6.5 2 12s4.5 10 10 10 10-4.5 10-10S17.5 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
      />
    </svg>
    <div
      className=""
    >
      <div
        className="c3"
      >
        <h5
          className="c4"
        />
        <span
          className="c5"
          fontSize={1}
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Banner renders with lightGreen bg 1`] = `
.c0 {
  color: #060;
  background-color: #ecf7ec;
  text-align: left;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c3 {
  text-align: left;
}

.c5 {
  font-size: 14px;
}

.c2 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  margin-right: 8px;
  margin-top: -2px;
}

.c4 {
  font-size: 24px;
  font-weight: 700;
  line-height: 1.25;
  margin: 0px;
}

<div
  className="c0"
  color="darkGreen"
>
  <div
    className="c1"
  >
    <svg
      className="c2"
      fill="currentcolor"
      height={24}
      viewBox="0 0 24 24"
      width={24}
    >
      <path
        d="M12 2C6.5 2 2 6.5 2 12s4.5 10 10 10 10-4.5 10-10S17.5 2 12 2zm-1.7 14.5L6 12.2 7.2 11l3.1 3.1 6.5-6.5L18 8.8l-7.7 7.7z"
      />
    </svg>
    <div
      className=""
    >
      <div
        className="c3"
      >
        <h5
          className="c4"
        />
        <span
          className="c5"
          fontSize={1}
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Banner renders with lightRed bg 1`] = `
.c0 {
  color: #800;
  background-color: #fbebeb;
  text-align: left;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c3 {
  text-align: left;
}

.c5 {
  font-size: 14px;
}

.c2 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  margin-right: 8px;
  margin-top: -2px;
}

.c4 {
  font-size: 24px;
  font-weight: 700;
  line-height: 1.25;
  margin: 0px;
}

<div
  className="c0"
  color="darkRed"
>
  <div
    className="c1"
  >
    <svg
      className="c2"
      fill="currentcolor"
      height={24}
      viewBox="0 0 24 24"
      width={24}
    >
      <path
        d="M12 2C6.5 2 2 6.5 2 12s4.5 10 10 10 10-4.5 10-10S17.5 2 12 2zm1 11h-2V7h2v6zm0 4h-2v-2h2v2z"
      />
    </svg>
    <div
      className=""
    >
      <div
        className="c3"
      >
        <h5
          className="c4"
        />
        <span
          className="c5"
          fontSize={1}
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Banner renders with no props other than theme 1`] = `
.c0 {
  color: #fff;
  background-color: #0a0;
  text-align: left;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c3 {
  text-align: left;
}

.c5 {
  font-size: 14px;
}

.c2 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  margin-right: 8px;
  margin-top: -2px;
}

.c4 {
  font-size: 24px;
  font-weight: 700;
  line-height: 1.25;
  margin: 0px;
}

<div
  className="c0"
  color="white"
>
  <div
    className="c1"
  >
    <svg
      className="c2"
      fill="currentcolor"
      height={24}
      viewBox="0 0 24 24"
      width={24}
    >
      <path
        d="M12 2C6.5 2 2 6.5 2 12s4.5 10 10 10 10-4.5 10-10S17.5 2 12 2zm-1.7 14.5L6 12.2 7.2 11l3.1 3.1 6.5-6.5L18 8.8l-7.7 7.7z"
      />
    </svg>
    <div
      className=""
    >
      <div
        className="c3"
      >
        <h5
          className="c4"
        />
        <span
          className="c5"
          fontSize={1}
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Banner renders with orange bg 1`] = `
.c0 {
  color: #fff;
  background-color: #f68013;
  text-align: left;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c3 {
  text-align: left;
}

.c5 {
  font-size: 14px;
}

.c2 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  margin-right: 8px;
  margin-top: -2px;
}

.c4 {
  font-size: 24px;
  font-weight: 700;
  line-height: 1.25;
  margin: 0px;
}

<div
  className="c0"
  color="white"
>
  <div
    className="c1"
  >
    <svg
      className="c2"
      fill="currentcolor"
      height={24}
      viewBox="0 0 24 24"
      width={24}
    >
      <path
        d="M1 21.5h22l-11-19-11 19zm12-3h-2v-2h2v2zm0-4h-2v-4h2v4z"
      />
    </svg>
    <div
      className=""
    >
      <div
        className="c3"
      >
        <h5
          className="c4"
        />
        <span
          className="c5"
          fontSize={1}
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Banner renders with red bg 1`] = `
.c0 {
  color: #fff;
  background-color: #c00;
  text-align: left;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c3 {
  text-align: left;
}

.c5 {
  font-size: 14px;
}

.c2 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  margin-right: 8px;
  margin-top: -2px;
}

.c4 {
  font-size: 24px;
  font-weight: 700;
  line-height: 1.25;
  margin: 0px;
}

<div
  className="c0"
  color="white"
>
  <div
    className="c1"
  >
    <svg
      className="c2"
      fill="currentcolor"
      height={24}
      viewBox="0 0 24 24"
      width={24}
    >
      <path
        d="M12 2C6.5 2 2 6.5 2 12s4.5 10 10 10 10-4.5 10-10S17.5 2 12 2zm1 11h-2V7h2v6zm0 4h-2v-2h2v2z"
      />
    </svg>
    <div
      className=""
    >
      <div
        className="c3"
      >
        <h5
          className="c4"
        />
        <span
          className="c5"
          fontSize={1}
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Banner renders with text node 1`] = `
.c0 {
  color: #fff;
  background-color: #0a0;
  text-align: left;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c3 {
  text-align: left;
}

.c5 {
  font-size: 14px;
}

.c2 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  margin-right: 8px;
  margin-top: -2px;
}

.c4 {
  font-size: 24px;
  font-weight: 700;
  line-height: 1.25;
  margin: 0px;
}

<div
  className="c0"
  color="white"
>
  <div
    className="c1"
  >
    <svg
      className="c2"
      fill="currentcolor"
      height={24}
      viewBox="0 0 24 24"
      width={24}
    >
      <path
        d="M12 17.8l6.2 3.7-1.6-7L22 9.7l-7.2-.6L12 2.5 9.2 9.1 2 9.7l5.5 4.8-1.6 7 6.1-3.7z"
      />
    </svg>
    <div
      className=""
    >
      <div
        className="c3"
      >
        <h5
          className="c4"
        >
          Header
        </h5>
        <span
          className="c5"
          fontSize={1}
        >
          <div
            className=""
          >
            Text
          </div>
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Banner renders with text string 1`] = `
.c0 {
  color: #fff;
  background-color: #0a0;
  text-align: left;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c3 {
  text-align: left;
}

.c5 {
  font-size: 14px;
}

.c2 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  margin-right: 8px;
  margin-top: -2px;
}

.c4 {
  font-size: 24px;
  font-weight: 700;
  line-height: 1.25;
  margin: 0px;
}

<div
  className="c0"
  color="white"
>
  <div
    className="c1"
  >
    <svg
      className="c2"
      fill="currentcolor"
      height={24}
      viewBox="0 0 24 24"
      width={24}
    >
      <path
        d="M12 17.8l6.2 3.7-1.6-7L22 9.7l-7.2-.6L12 2.5 9.2 9.1 2 9.7l5.5 4.8-1.6 7 6.1-3.7z"
      />
    </svg>
    <div
      className=""
    >
      <div
        className="c3"
      >
        <h5
          className="c4"
        >
          Header
        </h5>
        <span
          className="c5"
          fontSize={1}
        >
          Text
        </span>
      </div>
    </div>
  </div>
</div>
`;
